lib/moleculer/packets/info.rb

```diff
@@ -64,34 +64,37 @@ def as_json
       # @options data [Array<String>] ip_list the list of ip addresses for the node
       # @options data [String] hostname the hostname  of the node
       # @options data [Hash] client the client data for the node
-      def initialize(data)
-        super(data)
+      def initialize(config, data = {})
+        super(config, data)
         @services = HashUtil.fetch(data, :services)
-        @config   = OpenStruct.new(Hash[HashUtil.fetch(data, :config).map { |i| [StringUtil.underscore(i[0]), i[1]] }])
         @ip_list  = HashUtil.fetch(data, :ip_list)
         @hostname = HashUtil.fetch(data, :hostname)
-        @client   = Client.new(HashUtil.fetch(data, :client))
-        @node     = HashUtil.fetch(data, :node, nil)
+        @client   = Client.new(HashUtil.fetch(data, :client, {}))
+        node      = HashUtil.fetch(data, :node, nil)
+        @node_id  = HashUtil.fetch(data, :node_id, node&.id)
       end
 
       def topic
-        if @node
-          return "#{super}.#{@node.id}" if @node.is_a? Moleculer::Node
+        return "#{super}.#{@node_id}" if @node_id
 
-          return "#{super}.#{@node}"
-        end
         super
       end
 
-      def as_json
+      def to_h
         super.merge(
           services: @services,
-          config:   @config.to_h,
+          config:   config_for_hash,
           ipList:   @ip_list,
           hostname: @hostname,
-          client:   @client.as_json,
+          client:   @client.to_h,
         )
       end
+
+      private
+
+      def config_for_hash
+        Hash[config.to_h.to_a.reject { |a| a[0] == :log_file }]
```

Comment:
you don't have to create a new hash here or convert the config hash to an array before the block. This could be:

```ruby
def config_for_hash
  config.to_h.reject { |k, _v| k == :log_file }
end
```